compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1331,6 +1332,12 @@ impl<'a> Parser<'a> {
             return Ok(self.mk_await_expr(self_arg, lo));
         }
 
+        if self.eat_keyword(exp!(Use)) {
+            let use_span = self.prev_token.span;
+            self.psess.gated_spans.gate(sym::ergonomic_clones, use_span);
```

Comment:
Is it normal to gate in the parser like this?

Reply:
Looks like it is, I am not sure what's most common

Reply:
From what I've seen in code it is and this is what @compiler-errors what asking me to do in a previous review. Can you confirm Michael?.

Reply:
Not only is it "normal", it's literally *imperative* to gate in the parser to make sure that we don't accidentally stabilize the syntax in cfg'd code. Please make sure that there is a test that validates that this still gives an error when the code has `#[cfg(any())]`.